components/script/script_thread.rs

```diff
@@ -1654,7 +1654,17 @@ impl ScriptThread {
             .documents
             .borrow()
             .iter()
+            .filter(|(_, document)| document.window().is_top_level())
             .map(|(id, document)| (id, DomRoot::from_ref(&*document)))
+            .flat_map(|(id, document)| {
+                let mut documents = vec![(id.clone(), document.clone())];
+                documents.extend(document.iter_iframes().filter_map(|iframe| {
+                    iframe
+                        .pipeline_id()
```

Comment:
Note that our implementation of iframe is broken, and that https://github.com/servo/servo/blob/66edef806579fd0b386f4ceba473e6a9f7d0ca34/components/script/dom/htmliframeelement.rs#L462 likely returns `None` when it shouldn't. 

Perhaps a solution is to put the documents with not pipeline ids as last. (you can get the docs from the `documents` above).

Reply:
@gterzian found out that timeout it caused by `.filter(|(_, document)| !document.window().is_top_level())` not the filter_map

Reply:
this is result after removing `is_top_level`, not timeout but wrong result:

![image](https://github.com/servo/servo/assets/14943106/958bd8d6-7060-4179-8d71-83e878a30585)

Reply:
Ok good, sounds like reversing `iter_iframes` is the way to go then!

Reply:
@gterzian did you have time to look at this?